care/abdm/utils/api_call.py

```diff
@@ -0,0 +1,771 @@
+                        }
+                    ],
+                },
+            },
+        }
+
+        response = self.api.post(path, payload, None, additional_headers)
+        return response
+
+    def on_discover(self, data):
+        path = "/v0.5/care-contexts/on-discover"
+        additional_headers = {"X-CM-ID": settings.X_CM_ID}
+
+        request_id = str(uuid.uuid4())
+        payload = {
+            "requestId": request_id,
+            "timestamp": str(
+                datetime.now(tz=timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.000Z")
+            ),
+            "transactionId": data["transaction_id"],
+            "patient": {
+                "referenceNumber": data["patient_id"],
+                "display": data["patient_name"],
+                "careContexts": list(
+                    map(
+                        lambda context: {
+                            "referenceNumber": context["id"],
+                            "display": context["name"],
+                        },
+                        data["care_contexts"],
+                    )
+                ),
+                "matchedBy": data["matched_by"],
+            },
+            # "error": {"code": 1000, "message": "string"},
+            "resp": {"requestId": data["request_id"]},
+        }
+
+        response = self.api.post(path, payload, None, additional_headers)
+        return response
+
+    def on_link_init(self, data):
+        path = "/v0.5/links/link/on-init"
+        additional_headers = {"X-CM-ID": settings.X_CM_ID}
+
+        request_id = str(uuid.uuid4())
+        payload = {
+            "requestId": request_id,
+            "timestamp": str(
+                datetime.now(tz=timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.000Z")
+            ),
+            "transactionId": data["transaction_id"],
+            "link": {
+                "referenceNumber": data["patient_id"],
+                "authenticationType": "DIRECT",
+                "meta": {
+                    "communicationMedium": "MOBILE",
+                    "communicationHint": data["phone"],
+                    "communicationExpiry": str(
+                        (datetime.now() + timedelta(minutes=15)).strftime(
+                            "%Y-%m-%dT%H:%M:%S.000Z"
+                        )
+                    ),
+                },
+            },
+            # "error": {"code": 1000, "message": "string"},
+            "resp": {"requestId": data["request_id"]},
+        }
+
+        response = self.api.post(path, payload, None, additional_headers)
+        return response
+
+    def on_link_confirm(self, data):
+        path = "/v0.5/links/link/on-confirm"
+        additional_headers = {"X-CM-ID": settings.X_CM_ID}
+
+        request_id = str(uuid.uuid4())
+        payload = {
+            "requestId": request_id,
+            "timestamp": str(
+                datetime.now(tz=timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.000Z")
```

Comment:
strftime returns string so no need to cast it to to str again
^ more in this file